src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/OnBehalfCheckinSubmissionCoordinator.swift

```diff
@@ -0,0 +1,244 @@
+				self?.parentViewController.dismiss(animated: true)
+			}
+		)
+
+		let footerViewController = FooterViewController(
+			FooterViewModel(
+				primaryButtonName: AppStrings.OnBehalfCheckinSubmission.TraceLocationSelection.primaryButtonTitle,
+				isPrimaryButtonEnabled: true,
+				isSecondaryButtonHidden: true,
+				backgroundColor: .enaColor(for: .background)
+			)
+		)
+
+		let containerViewController = TopBottomContainerViewController(
+			topController: traceLocationSelectionViewController,
+			bottomController: footerViewController
+		)
+
+		navigationController.pushViewController(containerViewController, animated: true)
+	}
+
+	private func showQRCodeScanner() {
+		let qrCodeScanner = CheckinQRCodeScannerViewController(
+			qrCodeVerificationHelper: QRCodeVerificationHelper(),
+			appConfiguration: appConfiguration,
+			didScanCheckin: { [weak self] traceLocation in
+				self?.navigationController.dismiss(animated: true) {
+					self?.showDateTimeSelectionSelectionScreen(traceLocation: traceLocation)
+				}
+			},
+			dismiss: { [weak self] in
+				self?.traceLocationSelectionViewModel?.updateForCameraPermission()
+				self?.navigationController.dismiss(animated: true)
+			}
+		)
+
+		qrCodeScanner.definesPresentationContext = true
+
+		let modalNavigationController = UINavigationController(rootViewController: qrCodeScanner)
+		modalNavigationController.modalPresentationStyle = .fullScreen
+
+		navigationController.present(modalNavigationController, animated: true)
+	}
+
+	private func showDateTimeSelectionSelectionScreen(
+		traceLocation: TraceLocation
+	) {
+		let dateTimeSelectionViewController = OnBehalfDateTimeSelectionViewController(
+			traceLocation: traceLocation,
+			onPrimaryButtonTap: { [weak self] checkin in
+				self?.showTANInputScreen(
+					checkin: checkin
+				)
+			},
+			onDismiss: { [weak self] in
+				self?.parentViewController.dismiss(animated: true)
+			}
+		)
+
+		let footerViewController = FooterViewController(
+			FooterViewModel(
+				primaryButtonName: AppStrings.OnBehalfCheckinSubmission.TraceLocationSelection.primaryButtonTitle,
+				isPrimaryButtonEnabled: true,
+				isSecondaryButtonHidden: true,
+				backgroundColor: .enaColor(for: .background)
+			)
+		)
+
+		let containerViewController = TopBottomContainerViewController(
+			topController: dateTimeSelectionViewController,
+			bottomController: footerViewController
+		)
+
+		navigationController.pushViewController(containerViewController, animated: true)
+	}
+
+	private func showTANInputScreen(
+		checkin: Checkin
+	) {
+		let tanInputViewModel = TanInputViewModel(
+			title: AppStrings.OnBehalfCheckinSubmission.TANInput.title,
+			description: AppStrings.OnBehalfCheckinSubmission.TANInput.description,
+			onPrimaryButtonTap: { [weak self] tan, isLoading in
+				isLoading(true)
+
+				Log.info("[OnBehalfCheckinSubmission] Submitting with TAN \(private: tan)", log: .checkin)
+				DispatchQueue.main.asyncAfter(deadline: .now() + 1) { [weak self] in
```

Comment:
Setting `isLoading` to true for 1 second? Why is this happening instead of directly calling `showThankYouScreen`

Reply:
This is just for testing the UI implementation, this will be replaced with the actual request later on.